### DISCUSSION TITLE
Is AWS S3 client in scope of aws-sdk-rust?

### DISCUSSION THREAD

**Discussion by jeremychone** [awslabs/aws-sdk-rust#63] [Q&A] [answered]
UPDATE: Feel free to remove/hide this topic, since I just saw the S3 support as part of the roadmap. 

_Is S3 Client part of the scope of this SDK? or is there another official lib to look at. 

For now, it seems [rust-s3](https://crates.io/crates/rust-s3) is the most common._

**Comment by rcoh** [comment, marked as answer]
~~The S3 client is done yet but we're working on it!~~ 

The S3 client is done! It supports presigned URLs as well as standard operations (see examples)

**Reply by jeremychone** [reply]
What is your take on rust-s3 crate. I am testing it, looks pretty nice. Also, for us, while we deploy on Amazon S3, our dev definitely needs Minio for local dev. Hopefully, you will allow supporting custom endpoint and path strategy (as I think you do in the node.js one).

**Reply by franksl** [reply]
Hi,
We rely heavily on presigned GetObject and PutObject calls, will there be some ways to create presigned urls easily as done in other sdk's (php and node for sure)?
Thanks,
Frank

**Reply by rcoh** [reply]
The AWS SDK for Rust S3 client now supports presigned URLs
